### PR TITLE
Expose pretyping API needed by the language server

### PR DESCRIPTION
**Pull request by maximedenes**

### CODE REVIEW

File: compiler/src/pretyping.mli [hunk 1]
```diff
@@ -14,6 +14,14 @@ module Env : sig
   val add_from : 'asm env -> string * string -> 'asm env
   val dependencies : 'asm env -> string list list
 
+  val enter_file : 'asm env -> Annotations.pident option -> Location.t option -> string -> ('asm env * string) option
+  val exit_file : 'asm env -> 'asm env
+
+  module Funs : sig
+    val push : 'asm env -> (unit, 'asm) Prog.pfunc -> Prog.pty list -> 'asm env
+    val find : Annotations.symbol -> 'asm env -> ((unit, 'asm) Prog.pfunc * Prog.pty list) option
```
Comment (vbgl) [inline compiler/src/pretyping.mli:22]: The `Annotations.symbol` type is fishy. Should it just be `string`?

Comment (maximedenes) [reply compiler/src/pretyping.mli:22]: That's coming from the implementation (`pretyping.ml`). Do you mean that something is wrong there?

https://github.com/jasmin-lang/jasmin/blob/ec958d2a021a41a0a752e3bf2cc84dec8c91895b/compiler/src/pretyping.ml#L263

Comment (vbgl) [reply compiler/src/pretyping.mli:22]: Then everything’s right.

**Review by vbgl — APPROVED** [review]
I’ll merge unless CI fails.

**Comment by maximedenes** [comment]
Thanks!

**Pull request merged**